File: ios-app/Core/Models/Destination.swift
import Foundation

struct Destination: Codable, Hashable, Identifiable {
    let id: String
    let name: String
    let address: String
    let lat: Double
    let lng: Double
    let source: String
    let rawCategory: String?
    let normalizedCategory: String?

    init(
        id: String,
        name: String,
        address: String,
        lat: Double,
        lng: Double,
        source: String,
        rawCategory: String? = nil,
        normalizedCategory: String? = nil
    ) {
        self.id = id
        self.name = name
        self.address = address
        self.lat = lat
        self.lng = lng
        self.source = source
        self.rawCategory = rawCategory
        self.normalizedCategory = normalizedCategory
    }
}

struct DestinationSearchResponse: Codable {
    let items: [Destination]
}

File: shared-types/src/destination.ts
import type { PlaceCategory } from "./analytics.js";

export interface DestinationCandidate {
  id: string;
  name: string;
  address: string;
  lat: number;
  lng: number;
  source: "mock" | "kakao-local";
  rawCategory?: string | null;
  normalizedCategory?: PlaceCategory;
}

export interface DestinationSearchResponse {
  items: DestinationCandidate[];
}
